test/units/plugins/filter/test_ipaddr.py

```diff
@@ -502,3 +502,50 @@ def test_ipmath(self):
         with self.assertRaises(AnsibleFilterError) as exc:
             ipmath('1.2.3.4', 'some_number')
         self.assertEqual(exc.exception.message, expected)
+
+    def test_ipsubnet(self):
+        test_cases = (
+            (('1.1.1.1/24', '30'), '64'),
+            (('1.1.1.1/25', '24'), '0'),
+            (('1.12.1.34/32', '1.12.1.34/24'), '35'),
+            (('192.168.50.0/24', '192.168.0.0/16'), '51'),
+            (('192.168.144.5', '192.168.0.0/16'), '36870'),
+            (('192.168.144.5', '192.168.144.5/24'), '6'),
+            (('192.168.144.5/32', '192.168.144.0/24'), '6'),
+            (('192.168.144.1/30', '192.168.144.5/30'), False),
+            (('192.168.144.16/30', '192.168.144.0/24'), '5'),
+            (('192.168.144.5', ), '192.168.144.5/32'),
+            (('192.168.0.0/16', ), '192.168.0.0/16'),
+            (('192.168.144.5', ), '192.168.144.5/32'),
+            (('192.168.0.0/16', '20'), '16'),
+            (('192.168.0.0/16', '20', '0'), '192.168.0.0/20'),
+            (('192.168.0.0/16', '20', '-1'), '192.168.240.0/20'),
+            (('192.168.0.0/16', '20', '5'), '192.168.80.0/20'),
+            (('192.168.0.0/16', '20', '-5'), '192.168.176.0/20'),
+            (('192.168.144.5', '20'), '192.168.144.0/20'),
+            (('192.168.144.5', '18', '0'), '192.168.128.0/18'),
+            (('192.168.144.5', '18', '-1'), '192.168.144.4/31'),
+            (('192.168.144.5', '18', '5'), '192.168.144.0/23'),
+            (('192.168.144.5', '18', '-5'), '192.168.144.0/27'),
+            (('1.12.1.34/24', '1.12.1.34/32'), False),
+            (('span', 'test', 'error'), False),
+            (('test', ), False),            
+            (('192.168.144.5', '500000', '-5'), False),
+            (('192.168.144.5', '18', '500000'), False),
+            (('200000', '18', '-5'), '0.3.13.64/27'),
+        )
+        for args, res in test_cases:
+            self._test_ipsubnet(args, res)
+
+    def _test_ipsubnet(self, ipsubnet_args, expected_result):
+        self.assertEqual(ipsubnet(*ipsubnet_args), expected_result)
+
+        expected = 'An Exception indicating a network address is not correctly formatted.'
+        with self.assertRaises(AddrConversionError) as exc:
```

Comment:
I think, you're looking for [`self.assertRaisesRegexp(AddrConversionError, expected)`](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaisesRegexp)

Reply:
I don't find case to raise an error